pkg/archive/archive.go

```diff
@@ -534,6 +534,10 @@ func (ta *tarAppender) addTarFile(path, name string) error {
 	if ta.ChownOpts != nil {
 		hdr.Uid = ta.ChownOpts.UID
 		hdr.Gid = ta.ChownOpts.GID
+		// Don’t expose the user names from the local system; they probably don’t match the ta.ChownOpts value anyway,
+		// and they unnecessarily give recipients of the tar file potentially private data.
+		hdr.Uname = ""
+		hdr.Gname = ""
```

Comment:
Not that that's a blocker, mind.

Reply:
@nalind That’s quite attractive, I think it would be a replacement for https://github.com/containers/buildah/pull/4181 (with better performance).

OTOH it would make the archives worse for “ordinary `tar`” operations — in that case matching by account names better captures the users’ intent than matching by UIDs, in case the tar file is moved to a machine with different ID allocations.

I don’t know what we can automatically tell which one the caller wants… maybe based on `ta.ChownOpts != nil || !ta.IDMappings.Empty()`?

Or maybe there are no “ordinary `tar`” callers of this codebase anyway?! There _are_ several callers which use this for non-container-layer purposes, but it seems possible that they are only confined to a single host (where the distinction is moot), to a remote client-server operation (where we don’t want any `Chown` to happen), or to cross-namespace operations with layer/container data (where the configured ID mappings is what matters).

Reply:
Oh, you make a good point about the intent being hard to discern for cases where the names are preferred over IDs.  That makes the "filter the output to tailor it" approach more attractive to me in general than trying to add an option field to cover every possible case.
Given the either/or nature of the Uid/Uname and Gid/Gname fields, though, this would be a good exception to such a rule, so LGTM.

Reply:
Conceptually, I would prefer for `layerStore.Diff` and the like to return already-scrubbed tar files (in all situations with all graph drivers); cleaning that only later in Buildah works but feels a bit unclean to me.

The downside is that that ~silently changing the behavior of `pkg/{chroot,}archive` is a hard-to-manage API transition; at least it would require a detailed analysis of all users (most of which I’m unfamiliar with, notably the Podman checkpoint code).